src/rustcore/src/acceptors.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2019 Twitter, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::io::Result;
+use std::net::SocketAddr;
+
+use tokio::net::{TcpListener, TcpStream};
+use tokio::sync::mpsc::Sender;
+
+pub async fn tcp_acceptor(addr: SocketAddr, mut chan: Sender<TcpStream>) -> Result<()> {
+    let mut listener = TcpListener::bind(addr).await?;
+
+    loop {
+        let stream: TcpStream = match listener.accept().await {
+            Ok((stream, _)) => stream,
+            Err(e) => {
+                error!("Failed to establish connection: {}", e);
```

Comment:
technically the connection is already established by the time it gets to `accept()` stage, as the handshake is done in the kernel. Here, the error indicates there's nothing to accept (the syscall should only have one error type which is `EAGAIN`).

Reply:
In that case the error here should never happen since tokio should deal with `EAGAIN`. Not sure what error message to put here in that case.

Reply:
`"Failed to accept connection: {}"` might be the most descriptive thing here?

Reply:
I've modified this to `"Failed to accept connection: {}"` and made this debug level as suggested further down.